packages/datagateway-dataview/src/views/table/datafileTable.component.tsx

```diff
@@ -105,10 +105,18 @@ const DatafileTable = (props: DatafileTableProps): React.ReactElement => {
     [fetchNextPage]
   );
 
-  const aggregatedData: Datafile[] = React.useMemo(
-    () => (data ? ('pages' in data ? data.pages.flat() : data) : []),
-    [data]
-  );
+  /* istanbul ignore next */
+  const aggregatedData: Datafile[] = React.useMemo(() => {
+    if (data) {
+      if ('pages' in data) {
+        return data.pages.flat();
+      } else if (data instanceof Array) {
+        return data;
+      }
+    }
+
+    return [];
+  }, [data]);
```

Comment:
So to sum up everything that has happened in this PR:

1. build was failing because this newer version of Typescript seemed to be stricter on the type given to `aggregatedData`:

![agg data error](https://user-images.githubusercontent.com/71134574/182388177-e6217e2f-fb6f-407d-a77b-c29b811cfbe6.PNG)

I'm guessing the whole reason `'pages' in data ? data.pages.flat() : data` was there was to account for unit tests which mock the `useQuery` return type as an array.

2. I corrected this with the above method, which worked and everything passed. But then, codecov complained because we were getting only 60% coverage due to more lines being added and unit tests only covering some of them.
3. Rather than add a bunch of new unit tests to ensure coverage for this pointless thing, I thought I'd just ignore the entire function from the coverage altogether. As you can see with the latest build, this works.

@louise-davies would you mind verifying this solution before I merge it?

Reply:
Damn, didn't mean to comment yet. This assumes everything passes fine